--- a/config.mk
+++ b/config.mk
@@ -33,9 +33,9 @@ endif
 COMPAT_VERSIONS := $(shell I=$(COMPAT_LATEST_VERSION); while [ "$$I" -gt $(KERNEL_SUBLEVEL) ]; do echo $$I; I=$$(($$I - 1)); done)
 $(foreach ver,$(COMPAT_VERSIONS),$(eval CONFIG_COMPAT_KERNEL_3_$(ver)=y))
 
-ifdef CONFIG_COMPAT_KERNEL_2_6_24
-$(error "ERROR: compat-wireless by default supports kernels >= 2.6.24, try enabling only one driver though")
-endif #CONFIG_COMPAT_KERNEL_2_6_24
+ifdef CONFIG_COMPAT_KERNEL_2_6_22
+$(error "ERROR: compat-wireless by default supports kernels >= 2.6.22, try enabling only one driver though")
+endif #CONFIG_COMPAT_KERNEL_2_6_22
 
 ifeq ($(CONFIG_CFG80211),y)
 $(error "ERROR: your kernel has CONFIG_CFG80211=y, you should have it CONFIG_CFG80211=m if you want to use this thing.")
--- a/scripts/gen-compat-autoconf.sh
+++ b/scripts/gen-compat-autoconf.sh
@@ -11,7 +11,7 @@
 
 # This indicates which is the oldest kernel we support
 # Update this if you are adding support for older kernels.
-OLDEST_KERNEL_SUPPORTED="2.6.24"
+OLDEST_KERNEL_SUPPORTED="2.6.22"
 COMPAT_RELEASE=".compat_version"
 KERNEL_RELEASE=".compat_base_tree_version"
 MULT_DEP_FILE=".compat_pivot_dep"
