--- a/drivers/net/wireless/iwlegacy/common.c
+++ b/drivers/net/wireless/iwlegacy/common.c
@@ -540,6 +540,7 @@ il_led_brightness_set(struct led_classde
 	il_led_cmd(il, on, 0);
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25))
 static int
 il_led_blink_set(struct led_classdev *led_cdev, unsigned long *delay_on,
 		 unsigned long *delay_off)
@@ -548,6 +549,7 @@ il_led_blink_set(struct led_classdev *le
 
 	return il_led_cmd(il, *delay_on, *delay_off);
 }
+#endif
 
 void
 il_leds_init(struct il_priv *il)
@@ -561,7 +563,9 @@ il_leds_init(struct il_priv *il)
 	il->led.name =
 	    kasprintf(GFP_KERNEL, "%s-led", wiphy_name(il->hw->wiphy));
 	il->led.brightness_set = il_led_brightness_set;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25))
 	il->led.blink_set = il_led_blink_set;
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,30)
 	il->led.max_brightness = 1;
 #endif
--- a/drivers/net/wireless/iwlwifi/iwl-led.c
+++ b/drivers/net/wireless/iwlwifi/iwl-led.c
@@ -163,6 +163,7 @@ static void iwl_led_brightness_set(struc
 	iwl_led_cmd(priv, on, 0);
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25))
 static int iwl_led_blink_set(struct led_classdev *led_cdev,
 			     unsigned long *delay_on,
 			     unsigned long *delay_off)
@@ -171,6 +172,7 @@ static int iwl_led_blink_set(struct led_
 
 	return iwl_led_cmd(priv, *delay_on, *delay_off);
 }
+#endif
 
 void iwl_leds_init(struct iwl_priv *priv)
 {
@@ -187,7 +189,9 @@ void iwl_leds_init(struct iwl_priv *priv
 	priv->led.name = kasprintf(GFP_KERNEL, "%s-led",
 				   wiphy_name(priv->hw->wiphy));
 	priv->led.brightness_set = iwl_led_brightness_set;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25))
 	priv->led.blink_set = iwl_led_blink_set;
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,30)
 	priv->led.max_brightness = 1;
 #endif
